### PR TITLE
npm --access=restricted now supported

### PR DESCRIPTION
**Pull request by ahammond**
Default to access=public so we don't change existing behavior.

resolves https://github.com/awslabs/aws-delivlib/issues/368

-----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ahammond** [comment]
`npm run test` passed. Not sure why this failed.
Screen Shot 2020-06-11 at 1 47 12 PM

File: lib/publishing.ts [hunk 1]
```diff
@@ -105,6 +105,12 @@ export interface PublishToNpmProjectProps {
    * @default - npm default behavior ("latest" unless dist tag is specified in package.json)
    */
   distTag?: string;
+  /**
+   * npm --access public|restricted
+   *
+   * @default 'public'
+   */
+  access?: string;
 }
```
Comment (eladb) [inline lib/publishing.ts:114]: Can we use an enum here?

Comment (ahammond) [reply lib/publishing.ts:116]: Heck yes! :)

**Comment by ahammond** [comment]
@NetaNir any chance you could help me get some eyes on this?

File: lib/publishing.ts [hunk 2]
```diff
@@ -119,14 +127,17 @@ export class PublishToNpmProject extends cdk.Construct implements IPublisher {
 
     const forReal = props.dryRun === undefined ? 'false' : (!props.dryRun).toString();
 
+    const access = props.access || NpmAccess.PUBLIC;
```
Comment (eladb) [inline lib/publishing.ts:130]:
```suggestion
    const access = props.access ?? NpmAccess.PUBLIC;
```

File: lib/publishing.ts [hunk 3]
```diff
@@ -10,8 +10,10 @@ import { WritableGitHubRepo } from "./repo";
 import { LinuxPlatform, Shellable } from "./shellable";
 import { noUndefined } from "./util";
 
-
-
+export enum NpmAccess {
```
Comment (eladb) [inline lib/publishing.ts:13]: add docstrings on all public APIs

File: lib/publishing.ts [hunk 4]
```diff
@@ -105,6 +107,12 @@ export interface PublishToNpmProjectProps {
    * @default - npm default behavior ("latest" unless dist tag is specified in package.json)
    */
   distTag?: string;
+  /**
```
Comment (eladb) [inline lib/publishing.ts:110]:
```suggestion

    /**
```

**Comment by ahammond** [comment]
@eladb Thank you for the review! I've addressed comments.

**Comment by ahammond** [comment]
@ericzbeard this PR has been open for 2 months as of today. I've responded to feedback within a day. What else can I do to get this merged?

**Review by eladb — APPROVED** [review]

**Pull request merged**